src/libtest/lib.rs

```diff
@@ -601,6 +802,19 @@ pub fn parse_opts(args: &[String]) -> Option<OptRes> {
         };
     }
 
+    let report_time = unstable_optflag!(matches, allow_unstable, "report-time");
```

Comment:
This part from `let report_time` until `else { None };` can be extracted into a function.